virttest/qemu_vm.py

```diff
@@ -892,21 +892,25 @@ def add_memorys(devices, params):
             :param devices: VM devices container
             """
             options = []
-            params = params.object_params("mem")
-            if params.get("mem"):
+            mem_params = params.object_params("mem")
+            mem = float(mem_params.get("mem", 0))
+            maxmem = float(utils_misc.normalize_data_size(mem_params.get("maxmem", 0), order_magnitude="M"))
+            actual_mem = int(min(mem, maxmem))
+            params["mem"] = actual_mem
+            if mem_params.get("mem"):
                 options.append("%s" % params["mem"])
-            if params.get("slots") and params.get("maxmem"):
-                options.append("slots=%s" % params["slots"])
-                options.append("maxmem=%s" % params["maxmem"])
+            if mem_params.get("slots") and mem_params.get("maxmem"):
```

Comment:
why in line 897 ,you get it like mem_params.get("maxmem", 0) and in this line you invoke it mem_params.get("maxmem") , could you tell what is the difference ,thanks.